src/panel/PoiPanel.jsx

```diff
@@ -0,0 +1,237 @@
+/* globals _ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Telemetry from '../libs/telemetry';
+import nconf from '@qwant/nconf-getter';
+import layouts from './layouts.js';
+
+import ActionButtons from 'src/panel/poi/ActionButtons';
+import PoiHeader from 'src/panel/poi/PoiHeader';
+import PoiTitleImage from 'src/panel/poi/PoiTitleImage';
+import OpeningHour from 'src/components/OpeningHour';
+import OsmContribution from 'src/components/OsmContribution';
+import PoiBlockContainer from './poi_bloc/PoiBlockContainer';
+
+import CategoryService from '../adapters/category_service';
+import { openShareModal } from 'src/modals/ShareModal';
+
+import PoiStore from '../adapters/poi/poi_store';
+
+export default class PoiPanel extends React.Component {
+  static propTypes = {
+    poi: PropTypes.object.isRequired,
+    isFromFavorite: PropTypes.bool,
+    isFromCategory: PropTypes.bool,
+    sourceCategory: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+    this.poi = props.poi;
```

Comment:
You shouldn't do that in React, just use `this.props` directly, and if you need to compute things from props, do it in `render`. Otherwise this is the best way to have bugs from data inconsistency. For example, if `props` change with the component already mounted, which will happen all the time, the constructor won't be called so `this.poi` won't be kept up-to-date.